### PR TITLE
Add optimizer option to estimators

### PR DESCRIPTION
**Pull request by QRemy**
This PR add backend, optimize_opts, and covariance_opts options from`Fit.run()` to the `estimators.run()` methods`and their associated evaluate methods. 

Also set FluxEstimator._parameter_estimator as a property

### CODE REVIEW

**Comment by registerrier** [comment]
Thanks @QRemy . Having the possibility to modify the minimizer is indeed very useful. 
Maybe the arguments can be passed in the form of a dictionary, since they might evolve in the future?

File: gammapy/analysis/core.py [hunk 1]
```diff
@@ -207,6 +207,29 @@ def get_flux_points(self):
         cols = ["e_ref", "ref_flux", "dnde", "dnde_ul", "dnde_err", "is_ul"]
         log.info("\n{}".format(self.flux_points.data.table[cols]))
 
+    def get_excess_map(self):
```
Comment (adonath) [inline gammapy/analysis/core.py:210]: No sure why the is included here, must be a merge conflict...

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @QRemy, so far I don't have any comments. However please rebase and fix the remaining merge conflict...

**Review by adonath — APPROVED** [review]
Thanks @QRemy for the rebase! I'll merge this PR and then we can continue later this week...

**Pull request merged**